src/base.jl

```diff
@@ -1043,6 +1093,7 @@ end
 
 function handle_component_addition!(sys::System, bus::Bus)
     number = get_number(bus)
+    # FIXME: call `check_component_addition` here instead of `@assert`
```

Comment:
That already happens.  If this trips, it is a programmer mistake.

Reply:
Resolved in https://github.com/NREL/PowerSystems.jl/pull/390/commits/aa6ad0775b1ef4dff3e2cf013bccad77e9be45a6.